compact.go

```diff
@@ -280,6 +280,8 @@ func (c *LeveledCompactor) selectOverlappingDirs(ds []dirMeta) []string {
 				overlappingDirs = append(overlappingDirs, ds[i].dir)
 			}
 			overlappingDirs = append(overlappingDirs, d.dir)
+		}else if len(overlappingDirs) > 0 {
```

Comment:
is that right? Why we return only first overlap? Is this intented?

Reply:
hm you may have a look at the previous @codesome 's [comment](https://github.com/prometheus/tsdb/pull/560#issuecomment-476991384), if we return all, it can possibly result in compacting a very large block.